pyrs/calibration/peakfit_calibration.py

```diff
@@ -77,6 +78,51 @@ def __init__(self):
         return
 
 
+
+
+
+class Monosetting(Enum):
+    Si333 = ('Si333', 1.452)
+    Si511 = ('Si511', 1.452)
+    Si422 = ('Si422', 1.540)
+    Si331 = ('Si331', 1.731)
+    Si400 = ('Si400', 1.886)
+    Si311 = ('Si311', 2.275)
+    Si220 = ('Si220', 2.667)
+
+    def __new__(cls, label, wavelength):
+        obj = object.__new__(cls)
+        obj._value_ = label
+        obj.wavelength = wavelength
+        return obj
+
+    def __str__(self):
+        return self.value
+
+    def __float__(self):
+        return self.wavelength
+
+    @staticmethod
+    def getFromRotation(mrot):
+        if -41.0 < mrot < -38.0:
+            monosetting = Monosetting('Si333')
+        elif -1.0 < mrot < 1.0:
+            monosetting = Monosetting('Si511')
+        elif -20.0 < mrot < -19.0:
+            monosetting = Monosetting('Si422')
+        elif -170.0 < mrot < -166.0:
+            monosetting = MonoSetting('Si331')
+        elif 14.0 < mrot < 18.0:
+            monosetting = MonoSetting('Si400')
+        elif -11.0 < mrot < -8.0:
+            monosetting = MonoSetting('Si311')
+        elif -185.0 < mrot < -180.0:
+            monosetting = MonoSetting('Si220')
+        else:
+            raise ValueError('Unable to determine monosetting from the monochromator rotation angle')
```

Comment:
Add the value to the exception message so people know what specifically didn't work

Reply:
done